### ISSUE TITLE
Improve inference time. 

### ISSUE THREAD

**Issue by rohaantahir** [TencentARC/GFPGAN#334]
Hi, I love the work. I wanted to enhance the videos of containing facial images. To do that I am converting the video to frames and processing it one by one. I wanted to know if there any solution to improve the inferrence time, so that when a big video processed it can be processed efficiently. 

P.S I have RTX 3090 and core i9 high end system.

**Comment by Wong-denis** [comment]
same issue here. For me, it took about 90ms to process an 120x160 image. I wonder if there is room for improvement.

p.s. I use RTX3060.

**Comment by dearkafka** [comment]
@rohaantahir you know that 90ms is fast, right? Gfpgan is so far one of the fastest (still) given quality

**Comment by rohaantahir** [comment]
Yeah that seems fast, but, in my system I am getting morethan 90ms could you please tell me are you using. I am using this command to execute. 

python inference_gfpgan.py -i input_image -o  outputPath  -v 1.4 -s 2 --only_center_face --bg_upsampler none

**Comment by JGooLaaR** [comment]
@rohaantahir , you can skip frames without faces.

**Comment by rohaantahir** [comment]
@JGooLaaR whats the function of it ??

**Comment by rohaantahir** [comment]
no I am trying to figure it out.

**Comment by rohaantahir** [comment]
@Wong-denis what configurations you are using can you please tell me? like which pytorch, and python version specifically you are using?

**Comment by JGooLaaR** [comment]
@rohaantahir , you must use "Measure execution time of a function" way to find out what method is the bottleneck. Inside inference script.

**Comment by gg22mm** [comment]
> @rohaantahir, do you know the specific bottleneck code region?

![1732080231663](https://github.com/user-attachments/assets/9df064c7-e9da-414b-acc0-29fee5e3e3b8)
